### PR TITLE
KEB: Refactor pkg/errors --> errors

### PR DESCRIPTION
**Pull request by tillknuesting**
Refactor to use errors package from standard library

### CODE REVIEW

File: components/kyma-environment-broker/common/hyperscaler/shared_pool.go [hunk 1]
```diff
@@ -31,7 +30,7 @@ func (sp *sharedAccountPool) SharedCredentialsSecretBinding(hyperscalerType Type
 	labelSelector := fmt.Sprintf("shared=true,hyperscalerType=%s", hyperscalerType)
 	secretBindings, err := sp.getSecretBindings(labelSelector)
 	if err != nil {
-		return nil, errors.Wrap(err, "getting secret binding")
+		return nil, fmt.Errorf(" getting secret binding: %w", err)
```
Comment (wozniakjan) [inline components/kyma-environment-broker/common/hyperscaler/shared_pool.go:33]: nit: is the space intentional?
```suggestion
		return nil, fmt.Errorf("getting secret binding: %w", err)
```

File: components/kyma-environment-broker/internal/error/errors.go [hunk 2]
```diff
@@ -170,3 +170,27 @@ func checkK8SError(cause error) LastError {
 
 	return lastErr
 }
+
+// UnwrapOnce accesses the direct cause of the error if any, otherwise
+// returns nil.
+func UnwrapOnce(err error) (cause error) {
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/error/errors.go:176]: I know @PK85 said he is not interested in maintaining backward compatibility, but I want to advocate one last time we make sure not to break these two functions as there are metrics and alerts depending on them that may break
https://github.com/kyma-project/control-plane/blob/d5b5786ba3de3d53a74a80d291b7492060c2322d/components/kyma-environment-broker/internal/error/errors.go#L145
https://github.com/kyma-project/control-plane/blob/d5b5786ba3de3d53a74a80d291b7492060c2322d/components/kyma-environment-broker/internal/error/errors.go#L90

**Review by wozniakjan — COMMENTED** [review]
there are still some packages left, is it intentional?

<details>
    <summary>grep</summary>

```
$ grep . --exclude-dir=vendor -nre 'github.com/pkg/errors'
./cmd/broker/broker_suite_test.go:50:   "github.com/pkg/errors"
./cmd/broker/main.go:63:        "github.com/pkg/errors"
./cmd/environmentscleanup/main.go:18:   "github.com/pkg/errors"
./cmd/subscriptioncleanup/cloudprovider/azure.go:10:    "github.com/pkg/errors"
./cmd/subscriptioncleanup/cloudprovider/factory.go:7:   "github.com/pkg/errors"
./cmd/subscriptioncleanup/job/cleaner.go:10:    "github.com/pkg/errors"
./cmd/subscriptioncleanup/main.go:12:   "github.com/pkg/errors"
./cmd/subscriptioncleanup/model/types.go:3:import "github.com/pkg/errors"
./cmd/trialcleanup/main.go:14:  "github.com/pkg/errors"
./internal/appinfo/runtime_info.go:13:  "github.com/pkg/errors"
./internal/avs/client.go:15:    "github.com/pkg/errors"
./internal/avs/delegator.go:11: "github.com/pkg/errors"
./internal/broker/client.go:13: "github.com/pkg/errors"
./internal/broker/broker.go:11: "github.com/pkg/errors"
./internal/broker/instance_deprovision.go:11:   "github.com/pkg/errors"
./internal/broker/instance_last_operation.go:13:        "github.com/pkg/errors"
./internal/broker/services.go:8:        "github.com/pkg/errors"
./internal/broker/instance_create.go:24:        "github.com/pkg/errors"
./internal/edp/client.go:14:    "github.com/pkg/errors"
./internal/ias/client.go:14:    "github.com/pkg/errors"
./internal/ias/input.go:3:import "github.com/pkg/errors"
./internal/ias/bundle.go:8:     "github.com/pkg/errors"
./internal/process/deprovisioning/release_subscription_step_test.go:14: "github.com/pkg/errors"
./internal/process/provisioning/create_runtime_for_own_cluster_step.go:13:      "github.com/pkg/errors"
./internal/process/provisioning/create_runtime_without_kyma_step.go:14: "github.com/pkg/errors"
./internal/process/provisioning/initialisation.go:12:   "github.com/pkg/errors"
./internal/process/deprovision_operation.go:15: "github.com/pkg/errors"
./internal/process/input/input.go:17:   "github.com/pkg/errors"
./internal/process/input/builder.go:12: "github.com/pkg/errors"
./internal/process/provision_operation.go:15:   "github.com/pkg/errors"
./internal/process/update/upgrade_shoot_step.go:11:     "github.com/pkg/errors"
./internal/process/update_operation.go:11:      "github.com/pkg/errors"
./internal/process/upgrade_cluster_operation.go:11:     "github.com/pkg/errors"
./internal/process/upgrade_cluster_operation_test.go:7: "github.com/pkg/errors"
./internal/process/upgrade_kyma/initialisation.go:21:   "github.com/pkg/errors"
./internal/process/upgrade_kyma_operation.go:7: "github.com/pkg/errors"
./internal/process/operation_manager.go:11:     "github.com/pkg/errors"
./internal/storage/driver/postsql/operation.go:16:      "github.com/pkg/errors"
./internal/cis/client.go:12:    "github.com/pkg/errors"
./internal/cis/client_ver_1.go:11:      "github.com/pkg/errors"
./internal/cis/e2e/fake_broker_client.go:7:     "github.com/pkg/errors"
./internal/cis/account_cleanup.go:7:    "github.com/pkg/errors"
./internal/error/errors_test.go:13:     "github.com/pkg/errors"
./internal/error/not_found_error.go:3:import "github.com/pkg/errors"
./internal/error/temporary_error.go:6:  "github.com/pkg/errors"
./internal/error/temporary_error_test.go:7:     "github.com/pkg/errors"
./internal/error/errors.go:7:   "github.com/pkg/errors"
./internal/error/errors.go:186:// this is a replacement for github.com/pkg/errors.Cause
./go.mod:35:    github.com/pkg/errors v0.9.1
./go.sum:890:github.com/pkg/errors v0.8.0/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
./go.sum:891:github.com/pkg/errors v0.8.1-0.20171018195549-f15c970de5b7/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
./go.sum:892:github.com/pkg/errors v0.8.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
./go.sum:893:github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
./go.sum:894:github.com/pkg/errors v0.9.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
```
</details>

File: components/kyma-environment-broker/internal/error/errors.go (hunk 2 above)
Comment (tillknuesting) [reply components/kyma-environment-broker/internal/error/errors.go:176]: should be fine, but will check once again

File: components/kyma-environment-broker/internal/third_party/machinebox/graphql/graphql.go [hunk 3]
```diff
@@ -135,14 +133,18 @@ func (c *Client) runWithJSON(ctx context.Context, req *Request, resp interface{}
...
 	if _, err := io.Copy(&buf, res.Body); err != nil {
-		return errors.Wrap(err, "reading body")
+		return fmt.Errorf("copy body: %w", err)
 	}
 	c.logf("<< %s", buf.String())
 	if err := json.NewDecoder(&buf).Decode(&gr); err != nil {
 		if res.StatusCode != http.StatusOK {
 			return fmt.Errorf("graphql: server returned a non-200 status code: %v", res.StatusCode)
 		}
-		return errors.Wrap(err, "decoding response")
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/third_party/machinebox/graphql/graphql.go:145]: Note to my self: Double check this

File: components/kyma-environment-broker/internal/storage/postsql/init.go [hunk 4]
```diff
@@ -66,7 +64,7 @@ func CheckIfDatabaseInitialized(db *dbr.Connection) (bool, error) {
 			return false, nil
 		}
 
-		return false, errors.Wrap(err, "Failed to check if schema initialized")
+		return false, fmt.Errorf("failed to check if database is initialized: %w", err)
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/storage/postsql/init.go:67]: Note to my self: Change the text back

File: components/kyma-environment-broker/internal/orchestration/handlers/cluster_handler.go [hunk 5]
```diff
@@ -73,7 +73,8 @@ func (h *clusterHandler) createOrchestration(w http.ResponseWriter, r *http.Requ
 	err = h.orchestrations.Insert(o)
 	if err != nil {
 		h.log.Errorf("while inserting orchestration to storage: %v", err)
-		httputil.WriteErrorResponse(w, http.StatusInternalServerError, errors.Wrapf(err, "while inserting orchestration to storage"))
+		httputil.WriteErrorResponse(w, http.StatusInternalServerError, fmt.Errorf("while inserting orchestration to storage: %w", err))
+
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/orchestration/handlers/cluster_handler.go:77]: note to my self: remove blank line

File: components/kyma-environment-broker/internal/notification/client.go [hunk 6]
```diff
@@ -116,7 +114,8 @@ func (c *Client) callPost(path string, payload interface{}) (err error) {
 		}
 	}()
 	if err != nil {
-		return errors.Wrapf(err, "while making request for path %s", path)
+		return fmt.Errorf("while making request for path %s: %w", path, err)
+
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/notification/client.go:118]: note to my self: remove blank line

File: components/kyma-environment-broker/internal/ias/client.go [hunk 7]
```diff
@@ -175,12 +173,13 @@ func (c *Client) GenerateServiceProviderSecret(secretCfg SecretConfiguration) (_
...
 	}()
 	if err != nil {
-		return secretResponse, errors.Wrap(err, "while creating ServiceProvider secret")
+		return secretResponse, fmt.Errorf("while making request to generate ServiceProvider secret: %w", err)
 	}
 
 	err = json.NewDecoder(response.Body).Decode(secretResponse)
 	if err != nil {
-		return secretResponse, errors.Wrap(err, "while decoding response with secret provider")
+		//return secretResponse, errors.Wrap(err, "while decoding response with secret provider")
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/ias/client.go:181]: note to my self: remove comment

File: components/kyma-environment-broker/internal/error/not_found_error.go [hunk 8]
```diff
@@ -22,7 +20,7 @@ func (NotFoundError) Component() ErrComponent {
 }
 
 func IsNotFoundError(err error) bool {
-	cause := errors.Cause(err)
+	cause := UnwrapAll(err)
 	nfe, ok := cause.(interface {
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/error/not_found_error.go:24]: TODO: refactor

File: components/kyma-environment-broker/internal/error/errors_test.go [hunk 9]
```diff
@@ -81,7 +80,7 @@ func TestTemporaryErrorToLastError(t *testing.T) {
 			SetMessage(fmt.Sprintf("Got status %d", 502)).
 			SetReason(kebError.ErrHttpStatusCode).
 			SetComponent(kebError.ErrReconciler)
-		tempErr := errors.Wrap(kebError.WrapNewTemporaryError(errors.Wrap(err, "something")), "something else")
+		tempErr := fmt.Errorf("something: %w", kebError.WrapNewTemporaryError(fmt.Errorf("something: %w", err)))
```
Comment (tillknuesting) [inline components/kyma-environment-broker/internal/error/errors_test.go:83]: Note to my self: fix error message to something else

File: components/kyma-environment-broker/internal/error/errors.go [hunk 10]
```diff
@@ -92,7 +92,7 @@ func ReasonForError(err error) LastError {
 		return LastError{}
 	}
 
-	cause := errors.Cause(err)
+	cause := UnwrapAll(err)
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/error/errors.go:95]: @vvxxvvxx, it has been tested and this should for metric and alerting purposes be equivalent to the old error wrapping, but please keep an eye on any regressions and ping @kyma-project/gopher immediately if anything breaks.

**Review by wozniakjan — COMMENTED** [review]
/lgtm
/approve

a couple of minor nits, ping me when you have the conflicts resolved so we can merge it before it conflicts with anything else

File: components/kyma-environment-broker/internal/error/temporary_error_test.go [hunk 11]
```diff
@@ -15,9 +14,9 @@ func TestTemporaryError(t *testing.T) {
 	err3 := NewTemporaryError("some error: %s", fmt.Errorf("argErr"))
 
 	// when
-	e1 := errors.Wrapf(err1, "wrap err %s", "arg1")
-	e2 := AsTemporaryError(err2, "wrap err %s", "arg1")
-	e3 := errors.Wrapf(err3, "wrap err %s", "arg1")
+	e1 := fmt.Errorf("wrap err %s: %w", "arg1", err1)
+	e2 := fmt.Errorf("wrap err %s: %w", "arg1", err2)
```
Comment (vvxxvvxx) [inline components/kyma-environment-broker/internal/error/temporary_error_test.go:18]: seems this e2 is not same as previous `e2 := AsTemporaryError(err2, "wrap err %s", "arg1")`..

Comment (wozniakjan) [reply components/kyma-environment-broker/internal/error/temporary_error_test.go:18]: good catch, can you please take a look @tillknuesting?

**Review by wozniakjan — APPROVED** [review]

**Pull request merged**